taskweaver/planner/planner_prompt.yaml

```diff
@@ -96,44 +96,56 @@ instruction_template: |-
-    {planner_response_schema}
-  - Planner's response must include 5 elements "init_plan", "plan", "current_plan_step", "send_to", and "message".
+    {response_json_schema}
+  - Planner's response must include 5 attributes "init_plan", "plan", "current_plan_step", "send_to", and "message".
     - "init_plan" is the initial plan.
     - "plan" is the refined plan.
     - "current_plan_step" is the current step Planner is executing.
-    - "send_to" is the name of character (User or name of Worker) that Planner wants to speak to.
-    - "message" is the message of Planner sent to the receipt character. If there is any file path in the message, it should be formatted as links in Markdown, i.e., [file_name](file_path)
+    - "send_to" is the name of character (User or name of the Worker) that Planner wants to speak to.
+    - "message" is the message of Planner sent to the receipt Character. If there is any file path in the message, it should be formatted as links in Markdown, i.e., [file_name](file_path)
   - Planner must not include any other elements in the response that can cause parsing errors.
 
-planner_response_schema: |-
-  { 
-    "response": [
-      {
-        "type": "init_plan",
-        "content": "1. step 1\n2. step 2 <interactively or sequentially depend on 1>\n 3. step 3 <interactively or sequentially depend on 2>"
-      },
-      {
-        "type": "plan",
-        "content": "1. step 1 in the refined plan\n2. step 2 in the refined plan\n3. step 3 in the refined plan"
-      },
-      {
-        "type": "current_plan_step",
-        "content": "the current subtask that the Planner is executing"
-      },
-      {
-        "type": "send_to",
-        "content": "User or the name of Worker"
-      },
-      {
-        "type": "message",
-        "content": "The text message from the Planner"
-      }
-    ]
-  }
-
-
 experience_instruction: |-
   # Experience And Lessons
   Before starting planning, please refer to the following experiences and lessons learned from the previous tasks and include them in your plan.
   {experiences}
   You need to borrow the experience and lessons learned from the previous tasks in your current plan.
+
+response_json_schema: |-
+  {
+    "type": "object",
+    "properties": {
+        "response": {
+            "type": "object",
+            "properties": {
+                "init_plan": {
+                    "type": "string",
+                    "maxLength": 1000
+                },
+                "plan": {
+                    "type": "string",
+                    "maxLength": 1000
+                },
+                "current_plan_step": {
+                    "type": "string",
+                    "maxLength": 1000
+                },
+                "send_to": {
+                    "type": "string"
+                },
+                "message": {
+                    "type": "string",
+                    "maxLength": 1000
+                }
```

Comment:
Is the message with 1k character (rather than token) enough for all the cases?

Reply:
This is indeed a problem. I added this as some constrained generation methods have the problem of generating unlimited strings. Anyway, it is better to remove it.